src/guides/anatomy-of-an-http-transaction.md

```diff
@@ -0,0 +1,410 @@
+there are modules like [`body`][] on [`npm`][] which can help hide away some of
+this logic. It's important to have a good understanding of what's going on
+before going down that road, and that's why you're here!
+
+## A Quick Thing About Errors
+
+Since the `request` object is a [`ReadableStream`][], it's also an
+[`EventEmitter`][]. As such, it behaves in a slightly different way when an
+error happens.
+
+An error in the `request` stream presents itself by emitting an `'error'` event
+on the stream. **If you don't have a listener for that event, the error will be
+*thrown*, which could crash your Node.js program.** You should therefore add an
+`'error'` listener on your request streams, even if you just log it and
+continue on your way. (Though it's probably best to send some kind of HTTP error
+response. More on that later.)
+
+```javascript
+request.on('error', function(err) {
+  // This prints the error message and stack trace to `stderr`.
+  console.error(err.stack);
+});
+```
+
+There are other ways of [handling these errors][] such as using [`domains`][] or
+other abstractions and tools, but always be aware that errors can and do happen,
+and you're going to have to deal with them.
+
+## What We've Got so Far
+
+At this point, we've covered creating a server, and grabbing the method, URL,
+headers and body out of requests. When we put that all together, it might look
+something like this:
+
+```javascript
+var http = require('http');
+
+http.createServer(function(request, response) {
+  var headers = request.headers;
+  var method = request.method;
+  var url = request.url;
+  var body = '';
+  request.on('error', function(err) {
+    console.error(err);
+  }).on('data', function(chunk) {
+    body += chunk;
+  }).on('end', function() {
+    // At this point, we have the headers, method, url and body, and can now
+    // do whatever we need to in order to respond to this request.
+  });
+}).listen(80); // Activates this server, listening on port 80.
+```
+
+So far we haven't touched on the `response` object at all, which is an instance
+of [`ServerResponse`][], which is a [`WriteableStream`][]. It contains many
+useful methods for sending data back to the client. We'll cover that next.
+
+## HTTP Status Code
+
+If you don't bother setting it, the HTTP status code on a response will always
+be 200. Of course, not every HTTP response warrants this, and at some point
+you'll definitely want to send a different status code. To do that, you can set
+the `statusCode` property.
+
+```javascript
+response.statusCode = 404; // Tell the client that the resource wasn't found.
+```
+
+There are some other shortcuts to this, as we'll see soon.
+
+## Setting Response Headers
+
+Headers are set through a convenient method called [`setHeader`][].
+
+```javascript
+response.setHeader('Content-Type', 'application/json');
+response.setHeader('X-Powered-By', 'bacon');
+```
+
+When setting the headers on a response, the case is insensitive on their names.
```

Comment:
Based on your note about repeated headers on the request side, what happens if a header is set repeatedly?